### PR TITLE
iOS: Removed dependency on legacy Spotify SDK

### PR DESCRIPTION
**Pull request by manuelsc**
Plugin wasn't able to compile for iOS since the Spotify SDK it relies on isn't available anymore. The new sdk appears to have changed its API but I figured there's not need to even pull in that whole sdk dependency since we only need it for the OAuth URL.

I rewrote the code to work without this dependency, it should compile and work with iOS now.

This fixes Issue #28

### CODE REVIEW

File: src/ios/SpotifyOAuthPlugin.swift
Comment (NeoLegends) [inline src/ios/SpotifyOAuthPlugin.swift]: Could you convert that back to a guard? I feel that would be more idiomatic.

File: src/ios/SpotifyOAuthPlugin.swift
Comment (NeoLegends) [inline src/ios/SpotifyOAuthPlugin.swift]: I think there is a more idiomatic and fault-tolerant way to build URLs. Please see https://www.swiftbysundell.com/articles/constructing-urls-in-swift

File: src/ios/SpotifyOAuthPlugin.swift
Comment (NeoLegends) [inline src/ios/SpotifyOAuthPlugin.swift]: Instead of the for loop I would use the .joined(separator:) method here.

See https://stackoverflow.com/a/24826511

Comment (NeoLegends) [reply src/ios/SpotifyOAuthPlugin.swift]: Or rather, when using the idiomatic way of constructing the URL, this method would probably not be required anymore. :)

**Comment by tobika** [comment]
I'm only using the android auth part of this plugin but there is one advantage of using the sdk for authentication. If you have the Spotify app installed you don't need to enter your credentials and just click on a button to receive your token.

If I'm not mistaken with the web authentication you always have to enter your email/password.

**Review by NeoLegends — CHANGES_REQUESTED** [review]
Amazing changes! Thank you so much for this. You literally fixed the plugin. 

I have some changes addressing the way the URL is constructed. I think there is a more idiomatic and fault-tolerant way to do it instead of manually joining strings.

**Comment by manuelsc** [comment]
> I'm only using the android auth part of this plugin but there is one advantage of using the sdk for authentication. If you have the Spotify app installed you don't need to enter your credentials and just click on a button to receive your token.
> 
> If I'm not mistaken with the web authentication you always have to enter your email/password.

You are right I was thinking about that as well. But the current implementation only opens the webview on iOS as well. 
There appears to be a counterpart to spotifyWebAuthenticationURL namely spotifyAppAuthenticationURL that can be passed to UIApplication to open an installed Spotify app. But I don't know how this URL is constructed, maybe someone with the old spotify sdk installed can give us insight on that?

@NeoLegends 
I mostly work on Android which is why I'm not that fit with Swift. I'm sure there's a more idiomatic and cleaner approach. I'll try and look into the changes you suggested.

**Comment by NeoLegends** [comment]
@manuelsc Thank you for keeping up! Here is the link to the Documentation for the URL builder https://developer.apple.com/documentation/foundation/urlcomponents

File: src/ios/SpotifyOAuthPlugin.swift
Comment (manuelsc) [reply src/ios/SpotifyOAuthPlugin.swift]: I replaced the method with joined as suggested. I'm not sure about your second point so I'll leave this conversation open in case you have further feedback.

**Review by NeoLegends — APPROVED** [review]
@leolabs All good, I think. Merge?

**Comment by duffdean** [comment]
Is it possible to point my cordova plugins to this pull request? Or do we have to wait until the pull request is merged into the master?

**Pull request merged**
